docs/sdk/tutorials/plugins_development.md

```diff
@@ -0,0 +1,284 @@
+
+    def custom_method(self, project_id, label_id):
+        print(f"custom_method called for label {label_id}")
+        random_seed = np.random.random(1)[0]
+        if random_seed > 0.5:
+            self.logger.warning("Generating issue")
+            # Use kili for actions with self.kili
+            self.kili.append_to_issues(
+                label_id=label_id,
+                project_id=project_id,
+                text="Random issue generated for this label",
+            )
+
+    def on_review(self, label: Label, asset_id: str) -> None:
+        """
+        Dedicated handler for Review action
+        """
+        custom_function(label)
+
+    def on_submit(self, label: Label, asset_id: str) -> None:
+        """
+        Dedicated handler for Submit action
+        """
+        print("On submit called")
+
+        project_id = self.project_id
+        label_id = label.get("id")
+
+        self.custom_method(project_id, label_id)
+```
+
+### Testing the plugin locally
+
+In this we will show you how to test your plugin locally before uploading it.
+
+
+```python
+project_id = "<PROJECT ID>"
+```
+
+Instantiate the plugin:
+
+
+```python
+my_plugin_instance = PluginHandler(kili, project_id)
+
+
+def get_label(label_id, project_id):
+    """
+    Function to get the object Label with the same keys as it will be in the plugin
+    """
+    label = list(
+        kili.labels(
+            project_id=project_id,
+            label_id=label_id,
+            fields=["id", "jsonResponse", "author.id", "labelType", "createdAt", "secondsToLabel"],
+        )
+    )[0]
+
+    label["authorId"] = label["author"]["id"]
+    del label["author"]
+    return label
+```
+
+### Test the plugin run
+
+If you already have a test project with labels added, you can directly use the IDs of these labels (see the following cell). Otherwise, you can follow the *plugins_example.ipynb* notebook to create a new project and then upload an asset with an associated label.
+
+
+```python
+asset_id = "<YOUR_ASSET_ID>"
+label_id = "<YOUR_LABEL_ID>"
+```
+
+
+```python
+label = get_label(label_id=label_id, project_id=project_id)
+
+my_plugin_instance.on_submit(label=Label(**label), asset_id=asset_id)
+```
+
+
+
+    On submit called
+    custom_method called for label clcyqwn5a2gyk0lpn8d7w486h
+
+
+
+
+### Test the plugin run on Kili
+
+When you finish debugging the code, you may want to upload it directly into Kili.
+
+Note that you might get an error if the plugin name already exists in your Kili organization.
+
+
+```python
+path_to_plugin = "path/to/my/plugin.py"
+plugin_name = "My first kili plugin"
+```
+
+
+```python
+from kili.exceptions import GraphQLError
+
+try:
+    kili.upload_plugin(path_to_plugin, plugin_name)
+except GraphQLError as error:
+    print(str(error))
+```
+
+    Hint: A plugin with this name already exist, if you want to override it you can use the command kili.update_plugin(plugin_path="plugin.py", plugin_name="My first kili plugin")
```

Comment:
error here, we want to get rid of this isn't it ?

Reply:
I can silence the error, not sure what's best @xczuba

Reply:
@Jonas1312 @theodu If those are generated, is it really necessary to commit them ? ❓

Reply:
not necessary per se, but it's safer that way since the notebook to markdown conversion is still new and probably not bug free.

It's better to force the PR author to include the markdown and double-check the render in mkdocs